oci-unit-tests/loki_test.sh

```diff
@@ -0,0 +1,158 @@
+           --rm \
+           -d \
+           --name "${DOCKER_PREFIX}_${suffix}_promtail" \
+           "$@" \
+           --entrypoint "" \
+           "${DOCKER_IMAGE}" \
+           /usr/bin/promtail -config.file=/etc/loki/promtail-local-config.yaml
+}
+
+wait_loki_container_ready() {
+    local container="${1}"
+    local log="msg=\"Loki started\"$"
+    wait_container_ready "${container}" "${log}"
+}
+
+wait_promtail_container_ready() {
+    local container="${1}"
+    local log="msg=\"Starting Promtail\""
+    wait_container_ready "${container}" "${log}"
+}
+
+# Verify that loki is up and running.
+test_loki_up() {
+    debug "Creating loki container"
+
+    container=$(docker_run_loki -p "${LOKI_PORT}:${LOKI_PORT}")
+    assertNotNull "Failed to start the container" "${container}" || return 1
+    wait_loki_container_ready "${container}" || return 1
+
+    debug "Verifying that we can access the web endpoint"
+    local metrics
+    metrics="$(curl --silent "http://localhost:${LOKI_PORT}/metrics")"
+    assertTrue "Check if the cortex_query_frontend_retries_sum metric is present" "echo \"${metrics}\" | grep -qF cortex_query_frontend_retries_sum"
+    assertTrue "Check if the loki_boltdb_shipper_compact_tables_operation_duration_seconds metric is present" "echo \"${metrics}\" | grep -qF loki_boltdb_shipper_compact_tables_operation_duration_seconds"
+}
+
+# Verify that promtail and loki can talk to each other.
+test_loki_and_promtail() {
+    debug "Creating loki and promtail containers with custom config, making sure they talk to each other"
+
+    container=$(docker_run_loki \
+		-p "${LOKI_PORT}:${LOKI_PORT}" \
+		--network "${DOCKER_NETWORK}")
+    assertNotNull "Failed to start the container" "${container}" || return 1
+    wait_loki_container_ready "${container}" || return 1
+
+    local tempdir
+    tempdir=$(mktemp -d)
+    trap 'rm -rf ${tempdir}' 0 INT QUIT ABRT PIPE TERM
+    cat > "${tempdir}"/promtail-local-config.yaml << __EOF__
+server:
+  http_listen_port: 9080
+  grpc_listen_port: 0
+
+positions:
+  filename: /tmp/positions.yaml
+
+clients:
+  - url: http://${DOCKER_PREFIX}_${suffix}:3100/loki/api/v1/push
+
+scrape_configs:
+- job_name: system
+  static_configs:
+  - targets:
+      - localhost
+    labels:
+      job: varlogs
+      __path__: /var/log/*log
+__EOF__
+
+    local promtail_container
+    promtail_container=$(docker_run_promtail \
+		-p "${PROMTAIL_PORT}:${PROMTAIL_PORT}" \
+		--network "${DOCKER_NETWORK}" \
+		-v "${tempdir}"/promtail-local-config.yaml:/etc/loki/promtail-local-config.yaml)
+    assertNotNull "Failed to start the container" "${promtail_container}" || return 1
+    wait_promtail_container_ready "${promtail_container}" || return 1
+
+    # We have to sleep a bit and give some time for promtail to connect to loki.
+    sleep 5
+
+    local values
+    local match_re='{"status":"success","data":["varlogs"]}'
+    values=$(curl -s http://127.0.0.1:3100/loki/api/v1/label/job/values)
```

Comment:
Could we use $LOKI_PORT here (and below)?

Reply:
Good catch, thanks.  Fixed.